utils/celery_worker.py

```diff
@@ -83,96 +85,117 @@ def setup_periodic_tasks(sender, **kwargs):
+        logger.error('Task #{} failed'.format(args[2]))
+        logger.error('Traceback info:\n{}'.format(einfo))
 
-    resultlist = multiscanner.multiscan(list(files), configfile=config)
-    results = multiscanner.parse_reports(resultlist, python=True)
+        # Initialize the connection to the task DB
+        db.init_db()
 
-    scan_time = datetime.now().isoformat()
+        scan_time = datetime.now().isoformat()
 
-    # Loop through files in a way compatible with Py 2 and 3, and won't be
-    # affected by changing keys to original filenames
-    for file_ in files:
-        original_filename = files[file_]['original_filename']
-        task_id = files[file_]['task_id']
-        metadata = files[file_]['metadata']
-        # Get the Scan Config that the task was run with and
-        # add it to the task metadata
-        scan_config_object = configparser.SafeConfigParser()
-        scan_config_object.optionxform = str
-        scan_config_object.read(config)
-        full_conf = common.parse_config(scan_config_object)
-        sub_conf = {}
-        for key in full_conf:
-            if key == 'main':
-                continue
-            sub_conf[key] = {}
-            sub_conf[key]['ENABLED'] = full_conf[key]['ENABLED']
-        results[file_]['Scan Metadata'] = {}
-        results[file_]['Scan Metadata']['Worker Node'] = gethostname()
-        results[file_]['Scan Metadata']['Scan Config'] = sub_conf
-
-        # Use the original filename as the value for the filename
-        # in the report (instead of the tmp path assigned to the file
-        # by the REST API)
-        results[original_filename] = results[file_]
-        del results[file_]
-
-        results[original_filename]['Scan Time'] = scan_time
-        results[original_filename]['Metadata'] = metadata
-
-        # Update the task DB to reflect that the task is done
+        # Update the task DB with the failure
         db.update_task(
-            task_id=task_id,
-            task_status='Complete',
+            task_id=args[2],
+            task_status='Failed',
             timestamp=scan_time,
         )
 
-    # Save the reports to storage
-    storage_handler.store(results, wait=False)
-    storage_handler.close()
-
-    return results
-
 
-@app.task(base=Batches, flush_every=api_config['batch_size'], flush_interval=api_config['batch_interval'])
-def multiscanner_celery(requests, *args, **kwargs):
+@app.task(base=MultiScannerTask)
+def multiscanner_celery(file_, original_filename, task_id, file_hash, metadata, config=multiscanner.CONFIG, module_list=None):
     '''
-    Queue up multiscanner tasks and then run a batch of them at a time for
-    better performance.
+    Queue up multiscanner tasks
 
     Usage:
     from celery_worker import multiscanner_celery
     multiscanner_celery.delay(full_path, original_filename, task_id, metdata,
-                              hashed_filename, config)
+                              hashed_filename, metadata, config)
```

Comment:
This usage example needs to be cleaned up; `metdata` should be taken out and `module_list` should be added.